src/illuminatio/illuminatio_runner.py

```diff
@@ -153,92 +176,59 @@ def run_tests_for_target(network_ns, ports, target):
     LOGGER.info(svc_dns_entry)
     svc_ip = socket.gethostbyname(svc_dns_entry)
     LOGGER.info("Service IP: %s for Service: %s", svc_ip, target)
-    with tempfile.NamedTemporaryFile() as result_file:
-        LOGGER.debug("Results will be stored to %s", result_file)
-        # remove the need for nmap!
-        # e.g. https://gist.github.com/betrcode/0248f0fda894013382d7
-        # nmap that target TODO: handle None ip
-        # Replace bare nmap call with a better integrated solution like: https://pypi.org/project/python-nmap/ ?
-        nmap_cmd = ["nmap", "-oX", result_file.name, "-Pn", "-p", port_string, svc_ip]
-        LOGGER.info("running nmap with cmd %s", nmap_cmd)
-        prc = None
-        with Namespace(network_ns, "net"):
-            prc = subprocess.run(
-                nmap_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE
-            )
-        if prc is None or prc.returncode:
-            LOGGER.error("Executing nmap in foreign net ns failed! output:")
-            LOGGER.error(prc.stderr)
-            LOGGER.debug(prc)
-            return {
-                port_string: {
-                    "success": False,
-                    "error": "Couldn't nmap host %s with hostname %s"
-                    % (target, svc_dns_entry),
-                }
-            }
 
-        LOGGER.info("finished running nmap")
-        LOGGER.debug("Error log: %s", prc.stderr)
-        # when not using shell the output from nmap contains \\n instead of newline characters
-        LOGGER.info("Stdout: %s", str(prc.stdout).split("\\n"))
-        return extract_results_from_nmap_xml_file(result_file, port_on_nums, target)
+    ipv6_arg = ""
+    if ipaddress.ip_address(svc_ip).version == 6:
+        ipv6_arg = "-6"
 
+    # nmap that target TODO: handle None ip
```

Comment:
stale TODO? you already use this IP 3 lines above

Reply:
Let me check this tomorrow but I thing we still need to check for https://docs.python.org/3/library/socket.html#socket.herror in `svc_ip = socket.gethostbyname(svc_dns_entry)` but if we get `None` back we should already have an exception so yes this could be removed.